pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -122,6 +123,51 @@ func CreateOrLoadClusterInfo(context *clusterd.Context, namespace string, ownerR
 	return clusterInfo, maxMonID, monMapping, nil
 }
 
+// LoadExternalClusterExtraSecrets returns the secret value of the client.checker key
+func LoadExternalClusterExtraSecrets(context *clusterd.Context, namespace string) (cephconfig.ExternalCred, error) {
+	var externalCred cephconfig.ExternalCred
+
+	secret, err := context.Clientset.CoreV1().Secrets(namespace).Get(cephMonCheckerSecretName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return externalCred, errors.Wrapf(err, "failed to get checker secret")
+		}
+	}
+	// Populate external credential
+	externalCred.Username = string(secret.Data["userID"])
+	externalCred.Secret = string(secret.Data["userKey"])
+
+	_, err = context.Clientset.CoreV1().Secrets(namespace).Get(csi.CsiRBDNodeSecret, metav1.GetOptions{})
```

Comment:
suggestion: checking for the existence of the csi secrets seems like it should be a separate validation function instead of part of this function. Or maybe just name the method `ValidateAndLoadExternalClusterSecrets()`

Reply:
Renamed.